Error while wrapping message EventStore.Core.Messages.ClientMessage+ReadAllEventsForwardCompleted

Using EventStore 3.0.5 with the latest client lib from NuGet and get the following error message on a CatchUp Subscription in the ```$all``` stream using the ```SubscribeToAllFrom()``` method:

```
[PID:03564:014 2015.06.05 08:43:05.822 ERROR TcpDispatcher       ] Error while wrapping message EventStore.Core.Messages.ClientMessage+ReadAllEventsForwardCompleted.
System.NullReferenceException: Object reference not set to an instance of an object.
   at EventStore.Core.Messages.TcpClientMessageDto.EventRecord..ctor(EventRecord eventRecord) in c:\EventStore\src\EventStore.Core\Messages\TcpClientMessageDtoExtensions.cs:line 30
   at EventStore.Core.Messages.TcpClientMessageDto.ResolvedEvent..ctor(ResolvedEvent pair) in c:\EventStore\src\EventStore.Core\Messages\TcpClientMessageDtoExtensions.cs:line 19
   at EventStore.Core.Services.Transport.Tcp.ClientTcpDispatcher.ConvertToResolvedEvents(ResolvedEvent[] events) in c:\EventStore\src\EventStore.Core\Services\Transport\Tcp\ClientTcpDispatcher.cs:line 391
   at EventStore.Core.Services.Transport.Tcp.ClientTcpDispatcher.WrapReadAllEventsForwardCompleted(ReadAllEventsForwardCompleted msg) in c:\EventStore\src\EventStore.Core\Services\Transport\Tcp\ClientTcpDispatcher.cs:line 361
   at EventStore.Core.Services.Transport.Tcp.TcpDispatcher.WrapMessage(Message message) in c:\EventStore\src\EventStore.Core\Services\Transport\Tcp\TcpDispatcher.cs:line 64
```

Restarting the subscription again results in the same error message.

The subscription was working for the first 100.000 events and then stopped working with this error message.

The error of the client is:

```
2015-06-05 12:37:53.0145 DNA-DEV-WIN8 [ERROR] EventStoreConnection[7]: EventStoreConnection 'ES-c570d121-ed5f-445e-84ce-7ccfabc21b30': operation never got response from server.
UTC now: 10:37:53.011, operation: Operation ReadAllEventsForwardOperation ({7bb0be0d-1cfa-4201-8be3-2f4f1c737912}): Position: 641363/641363, MaxCount: 500, ResolveLinkTos: True, RequireMaster: True, retry count: 0, created: 10:37:45.055, last updated: 10:37:45.055. 
```

So it is at Position: 641363/641363? How do i get the event at this position? Via WebUI ?

Thats one way ... try browsing to $all and you can see how the uris work.

If not we have other tools that can look at the data.

On Fri, Jun 5, 2015 at 4:08 PM, Daniel Nauck <notifications@github.com>
wrote:

> The error of the client is:
>
> 2015-06-05 12:37:53.0145 DNA-DEV-WIN8 [ERROR] EventStoreConnection[7]: EventStoreConnection 'ES-c570d121-ed5f-445e-84ce-7ccfabc21b30': operation never got response from server.
> UTC now: 10:37:53.011, operation: Operation ReadAllEventsForwardOperation ({7bb0be0d-1cfa-4201-8be3-2f4f1c737912}): Position: 641363/641363, MaxCount: 500, ResolveLinkTos: True, RequireMaster: True, retry count: 0, created: 10:37:45.055, last updated: 10:37:45.055.
>
> So it is at Position: 641363/641363? How do i get the event at this
> position? Via WebUI ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EventStore/EventStore/issues/503#issuecomment-109288841>
> .
>



-- 
Studying for the Turing test

--- a/src/EventStore.ClientAPI/ResolvedEvent.cs
+++ b/src/EventStore.ClientAPI/ResolvedEvent.cs
@@ -50,7 +50,7 @@ namespace EventStore.ClientAPI
 
         internal ResolvedEvent(ClientMessage.ResolvedEvent evnt)
         {
-            Event = new RecordedEvent(evnt.Event);
+            Event = evnt.Event == null ? null : new RecordedEvent(evnt.Event);
             Link = evnt.Link == null ? null : new RecordedEvent(evnt.Link);
             OriginalPosition = new Position(evnt.CommitPosition, evnt.PreparePosition);
         }

--- a/src/EventStore.Core/Messages/TcpClientMessageDtoExtensions.cs
+++ b/src/EventStore.Core/Messages/TcpClientMessageDtoExtensions.cs
@@ -17,7 +17,7 @@ namespace EventStore.Core.Messages
         public partial class ResolvedEvent
         {
             public ResolvedEvent(Data.ResolvedEvent pair)
-                : this(new EventRecord(pair.Event),
+                : this(pair.Event != null ? new EventRecord(pair.Event) : null,
                        pair.Link != null ? new EventRecord(pair.Link) : null,
                        pair.OriginalPosition.Value.CommitPosition,
                        pair.OriginalPosition.Value.PreparePosition)

